content/part4.md

```diff
@@ -0,0 +1,199 @@
+In general, HTTP GET requests execute Queries, HTTP PUT or PATCH requests send Signals, and HTTP POST requests create new Workflows.
+
+```go
+// Create a new cart
+r.Handle("/cart", http.HandlerFunc(CreateCartHandler)).Methods("POST")
+// Get the state of an existing cart
+r.Handle("/cart/{workflowID}/{runID}", http.HandlerFunc(GetCartHandler)).Methods("GET")
+
+// Add a new item to the cart
+r.Handle("/cart/{workflowID}/{runID}/add", http.HandlerFunc(AddToCartHandler)).Methods("PUT")
+// Remove an item from the cart
+r.Handle("/cart/{workflowID}/{runID}/remove", http.HandlerFunc(RemoveFromCartHandler)).Methods("PUT")
+// Update the cart's associated email address
+r.Handle("/cart/{workflowID}/{runID}/email", http.HandlerFunc(UpdateEmailHandler)).Methods("PUT")
+// Check out
+r.Handle("/cart/{workflowID}/{runID}/checkout", http.HandlerFunc(CheckoutHandler)).Methods("PUT")
+```
+
+In this case, the API server and the [Worker](https://docs.temporal.io/docs/go/workers) are separate processes.
+The API server is just an intermediary between the Temporal server and your API server's clients.
+The carts themselves are stored in the Temporal server and in the Worker process.
+
+## Handler Functions
+
+First, let's take a look at the `POST /cart` endpoint. Since we've chosen to represent an individual shopping cart as a Workflow, the `CreateCartHandler()` function will create a new Workflow using `ExecuteWorkflow()`.
+You are responsible for making sure each `POST /cart` call creates a Workflow creates a unique `workflowID`.
+[Temporal treats a Workflow with the same ID as a prior Workflow as a continuation of the prior Workflow](https://docs.temporal.io/docs/shared/continue-as-new).
+
+```go
+func CreateCartHandler(w http.ResponseWriter, r *http.Request) {
+  // In production you should use uuids or something similar, but the
+  // current time is enough for this example. Make sure the Workflow ID
+  // is unique every time the user creates a new cart!
+	workflowID := "CART-" + fmt.Sprintf("%d", time.Now().Unix())
+
+	options := client.StartWorkflowOptions{
+		ID:        workflowID,
+		TaskQueue: "CART_TASK_QUEUE",
+	}
+
+	cart := app.CartState{Items: make([]app.CartItem, 0)}
+	we, err := temporal.ExecuteWorkflow(context.Background(), options, app.CartWorkflow, cart)
+	if err != nil {
+		WriteError(w, err)
+		return
+	}
+
+  // Return the `workflowID` and `runID` so clients can use them with other endpoints
+	res := make(map[string]interface{})
+	res["cart"] = cart
+	res["workflowID"] = we.GetID()
+	res["runID"] = we.GetRunID()
+
+	w.WriteHeader(http.StatusCreated)
+	json.NewEncoder(w).Encode(res)
+}
+```
+
+Now you have a `POST /cart` endpoint that creates a new empty cart, and returns the `workflowID` and `runID` that uniquely identify this Workflow.
+Even though `workflowID` is unique in this case, you still need to provide `runID` in order to Query or Signal the cart Workflow.
```

Comment:
That isn't true. By default those operations will use the most recent run id if one isn't provided.